components/x-teaser-timeline/src/lib/transform.js

```diff
@@ -135,11 +135,13 @@ const getItemGroups = ({
 const getGroupAndIndex = (groups, position) => {
 	if (position > 0) {
 		const group = groups.findIndex((g) => g.items.some((item) => item.articleIndex === position - 1))
-		const index = groups[group].items.findIndex((item) => item.articleIndex === position - 1)
+		if(groups[group]){
+			const index = groups[group].items.findIndex((item) => item.articleIndex === position - 1)
 
-		return {
-			group: group,
-			index: index + 1
+			return {
+				group: group,
+				index: index + 1
+			}
```

Comment:
what happens if `groups[group]` is undefined? I guess this function will return undefined and the exception will be thrown somewhere else? Have you tested it?

Reply:
If groups[group] is undefined then is returned:	
```
{
    group: 0,
    index: 0
}
```
This is a suggestion as I am not aware of the business logic; anyway, this is a simple way to avoid the TypeError.
Apparently, the only way to get groups[group] as undefined is that there isn't any match, so group (index of the match) is -1. Therefore groups[-1] is undefined.
Probably this code needs proper testing to make sure of it, but it's out of the scope of ft.com team, sorry! 😉

Reply:
If groups[group] is undefined then is returned:	
```
{
    group: 0,
    index: 0
}
```
This is a suggestion as I am not aware of the business logic; anyway, this is a simple way to avoid the TypeError.
Apparently, the only way to get groups[group] as undefined is that there isn't any match, so group (index of the match) is -1. Therefore groups[-1] is undefined.
Probably this code needs proper testing to make sure of it, but it's out of the scope of ft.com team, sorry! 😉